kimchi/src/proof.rs

```diff
@@ -181,6 +184,47 @@ impl<G: AffineCurve> RecursionChallenge<G> {
     pub fn new(chals: Vec<G::ScalarField>, comm: PolyComm<G>) -> RecursionChallenge<G> {
         RecursionChallenge { chals, comm }
     }
+
+    pub fn evals(
+        &self,
+        max_poly_size: usize,
+        evaluation_points: &[G::ScalarField],
+        powers_of_eval_points_for_chunks: &[G::ScalarField],
+    ) -> Vec<Vec<G::ScalarField>> {
+        let RecursionChallenge { chals, comm: _ } = self;
+        // No need to check the correctness of poly explicitly. Its correctness is assured by the
+        // checking of the inner product argument.
+        let b_len = 1 << chals.len();
+        let mut b: Option<Vec<G::ScalarField>> = None;
+
+        (0..2)
+            .map(|i| {
```

Comment:
(don't really dig the functional style here, it takes more lines and adds more indentation)

Reply:
ah I see now looking at the commits that it's just moving code around